### PR TITLE
Added in some view the IHProgressHUD

### PR DESCRIPTION
**Pull request by marinofaggiana**
Some view needs a better HUD with the progress, as the upload files in the `share extension`, so I have found this library that is not perfect (I don't like the `singleton`) but is a valid for the extension because can used the `view` and not only the `appDelegate.window?.rootViewController?`

### CODE REVIEW

File: Share/NCShareExtension.swift [hunk 1]
```diff
@@ -65,6 +66,9 @@ class NCShareExtension: UIViewController, NCListCellDelegate, NCEmptyDataSetDele
     private var activeAccount: tableAccount!
     private let chunckSize = CCUtility.getChunkSize() * 1000000
     
+    private var numberFilesName: Int = 0
```
Comment (thisIsTheFoxe) [inline Share/NCShareExtension.swift:69]: Is this being used for something else or is there another reason you can't just do:
```suggestion
    private var numberFilesName: Int { filesName.count }
```

File: Share/NCShareExtension.swift [hunk 2]
```diff
@@ -412,28 +435,33 @@ class NCShareExtension: UIViewController, NCListCellDelegate, NCEmptyDataSetDele
...
                     }
                 }
             }
         } else {
-            extensionContext?.completeRequest(returningItems: extensionContext?.inputItems, completionHandler: nil)
-            return
+            
+            IHProgressHUD.showSuccesswithStatus(NSLocalizedString("_success_", comment: ""))
+
+            DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
```
Comment (thisIsTheFoxe) [inline Share/NCShareExtension.swift:462]: 2s seems a bit arbitrary and personally a bit too long for me. If everything succeeded it should just dismiss right away - imo.

File: iOSClient/Main/NCFunctionCenter.swift [hunk 3]
```diff
@@ -352,34 +353,34 @@ import Queuer
         }
         
         if CCUtility.fileProviderStorageExists(metadata.ocId, fileNameView: metadata.fileNameView) && CCUtility.fileProviderStorageExists(metadataMOV.ocId, fileNameView: metadataMOV.fileNameView) {
-            saveLivePhotoToDisk(metadata: metadata, metadataMov: metadataMOV, progressView: nil, viewActivity: self.appDelegate.window?.rootViewController?.view)
+            saveLivePhotoToDisk(metadata: metadata, metadataMov: metadataMOV)
         }
     }
     
-    func saveLivePhotoToDisk(metadata: tableMetadata, metadataMov: tableMetadata, progressView: UIProgressView?, viewActivity: UIView?) {
+    func saveLivePhotoToDisk(metadata: tableMetadata, metadataMov: tableMetadata) {
```
Comment (thisIsTheFoxe) [inline iOSClient/Main/NCFunctionCenter.swift:360]: Again it seems a bit random why the change is made here and `UIProgressView` is still being used in other parts of the app..?

**Review by thisIsTheFoxe — CHANGES_REQUESTED** [review]
Thank, tho I'm not yet entirely sure why the change is necessary. 
Anyway, a few other UX concerns:
- using both `IHProgressHUD` and `UIProgressView` might make the app feel inconsistent
- when displaying the `IHProgressHUD` the entire view becomes unresponsive. Which feels unintuitive to me. If that's intended, may I suggest darkening the rest like system alerts do?

File: Share/NCShareExtension.swift (hunk 2 above)
Comment (marinofaggiana) [reply Share/NCShareExtension.swift:462]: true ... but some App(s) displays the final message (success) for a few seconds for comforts the user, other close immediataly the view .... it always depends on the point of view, ...

**Comment by marinofaggiana** [comment]
> Thank, tho I'm not yet entirely sure why the change is necessary. Anyway, a few other UX concerns:
> 
> * using both `IHProgressHUD` and `UIProgressView` might make the app feel inconsistent
> * when displaying the `IHProgressHUD` the entire view becomes unresponsive. Which feels unintuitive to me. If that's intended, may I suggest darkening the rest like system alerts do?

The hud is different from the UIProgressView and the use of one does not exclude the other as well as the unresponsive ... depends on where and how it is used.

File: Share/NCShareExtension.swift (hunk 2 above)
Comment (marinofaggiana) [reply Share/NCShareExtension.swift:461]: @Ivansss @alazarevic5 what's your opinion...

Comment (marinofaggiana) [reply Share/NCShareExtension.swift:461]: mmm nope ... @thisIsTheFoxe 0, 1 or 2 sec. ?

File: iOSClient/Main/NCFunctionCenter.swift (hunk 3 above)
Comment (marinofaggiana) [reply iOSClient/Main/NCFunctionCenter.swift:360]: It's present in Viewer image for the progress download image

File: Share/NCShareExtension.swift (hunk 2 above)
Comment (marinofaggiana) [reply Share/NCShareExtension.swift]: 1 for me the user must see the "success""

**Comment by marinofaggiana** [comment]
Ok we can try it
